src/types/UiaTextRangeBase.cpp

```diff
@@ -463,7 +463,7 @@ try
     *ppRetVal = nullptr;
 
 #pragma warning(suppress : 26447) // QueryInterface's exception should be caught by the try-CATCH_RETURN block to allow this to be noexcept
-    auto hr = _pProvider->QueryInterface(IID_PPV_ARGS(ppRetVal));
+    const auto hr = _pProvider->QueryInterface(IID_PPV_ARGS(ppRetVal));

```

Comment:
if `_pProvider` is a ComPtr here and you _are_ changing this line, please use `_pProvider.CopyTo(ppRetVal)`

Reply:
not a ComPtr :( just a weak pointer.